lib/spack/spack/test/ci.py

```diff
@@ -421,6 +421,7 @@ def make_yaml_obj(use_artifact_buildcache, optimize, use_dependencies):
 def test_get_spec_filter_list(mutable_mock_env_path, config, mutable_mock_repo):
     """Test that given an active environment and list of touched pkgs,
     we get the right list of possibly-changed env specs"""
+    spack.config.set("config:environments_root", str(tmpdir.join("local_envs")))
```

Comment:
This test doesn't include `tmpdir`, I'm surprised it would even run like this.

Reply:
It doesn't. This was a hacky attempt to get it to work based on what I did in a different test. For whatever reason the `mutable_mock_env` fixture's path is not correct for this test.